src/librustc_resolve/build_reduced_graph.rs

```diff
@@ -731,7 +733,7 @@ impl<'a, 'b> BuildReducedGraphVisitor<'a, 'b> {
                 self.parent_scope.module = module;
 
                 for variant in &(*enum_definition).variants {
-                    self.build_reduced_graph_for_variant(variant, vis);
+                    visit::walk_variant(self, variant);
                 }
```

Comment:
With `visit_variant` in place this whole loop can be removed (otherwise we are walking variants twice).